src/Core/Configurations/RuntimeConfigValidator.cs

```diff
@@ -652,8 +652,24 @@ public void ValidateRelationshipsInConfig(RuntimeConfig runtimeConfig, IMetadata
             string databaseName = runtimeConfig.GetDataSourceNameFromEntityName(entityName);
             ISqlMetadataProvider sqlMetadataProvider = sqlMetadataProviderFactory.GetMetadataProvider(databaseName);
 
+            // Dictionary to store mapping from target entity's name to relationship name. Whenever we encounter that we
+            // are getting more than 1 entry for a target entity, we throw a validation error as it indicates the user has
+            // defined multiple relationships between the same source and target entities.
+            Dictionary<string, string> targetEntityNameToRelationshipName = new();
             foreach ((string relationshipName, EntityRelationship relationship) in entity.Relationships!)
             {
+                string targetEntityName = relationship.TargetEntity;
+                if (targetEntityNameToRelationshipName.TryGetValue(targetEntityName, out string? duplicateRelationshipName))
+                {
+                    throw new DataApiBuilderException(
+                        message: $"Found multiple relationships: {duplicateRelationshipName}, {relationshipName} between source entity: {entityName} and target entity: {targetEntityName}",
```

Comment:
Perhaps, also mention that this is not supported and is an invalid config? The substatus code is really for the triaging if an issue is opened.